app/src/screen-components/user-news/news-list.tsx

```diff
@@ -0,0 +1,297 @@
+      alignSelf: "center",
+    }}
+  >
+    <MaterialCommunityIcons name={selected ? 'check' : 'plus'} color={Color["black-32"]} size={24} />
+  </TouchableOpacity>
+)
+
+interface ItemProps {
+  item: Article
+  onPress: (item: Article) => void
+  onPlusPress: (item: Article) => void
+}
+
+const Item: FC<ItemProps> = ({ item, onPress, onPlusPress }) => {
+  const theme = useTheme()
+  const fonts = useFonts()
+  const { state } = usePlayer()
+
+  const [inQueue, setInQueue] = useState(false)
+
+  useEffect(() => {
+    const queue = RNTrackPlayer.getQueue().then((queue) => {
+      const filtered = queue.filter((value, index, array) => {
+        return value.id == item.id
+      })
+      setInQueue(filtered.length > 0)
+    })
+
+  }, [state])
+
+
+  return (<View style={{ alignItems: "center" }}>
+    <View
+      style={{
+        flexDirection: "row",
+        justifyContent: "space-between",
+        paddingStart: 16,
+        paddingEnd: 16,
+        paddingBottom: 14,
+        paddingTop: 14
+      }}
+    >
+      <View style={{ width: 68, height: 50 }}>
+        <Image
+          style={{
+            width: "100%",
+            height: "100%",
+            borderRadius: 10,
+            backgroundColor: 'gray'
+          }}
+          source={{
+            uri: item.img,
+          }}
+        />
+      </View>
+      <TouchableOpacity
+        onPress={() => { onPress(item) }}
+        style={{
+          flex: 1,
+          marginStart: 16,
+          marginEnd: 8
+        }}
+      >
+        <View style={{}}>
+          <Text style={fonts.titleSmall} numberOfLines={2} >
+            {item.title}
+          </Text>
+          <Text style={StyleSheet.compose(fonts.textXSmall, { color: theme.colors.textLight })} >
+            {item.published}
+          </Text>
+        </View>
+      </TouchableOpacity>
+      <PlusIcon onPress={() => { onPlusPress(item) }} selected={inQueue} />
+    </View>
+  </View>)
+}
+
+const QUERY = gql`
+query Articles($first: Int, $after: String, $categories: [ID], $gteDate: Date, $lteDate: Date) {
+	articles(first: $first, after: $after, category_Key_In: $categories, pubDate_Gte: $gteDate, pubDate_Lte: $lteDate) {
+    pageInfo{
+      hasNextPage,
+      endCursor
+    },
+    edges{
+      node {
+        id,
+        title,
+        url: recordingUrl,
+        published: pubDate,
+        provider {
+          name
+        }
+      }
+    }
+  }
+}
+`
+
+export const getTimeRange = (timeRange: TimeRangeItem): TimeRange => {
+  var gteDate = new Date()
+  switch (timeRange) {
+    case TimeRangeItem.now:
+      gteDate.setHours(gteDate.getHours() - 2)
+      break
+    case TimeRangeItem.today:
+      gteDate.setHours(gteDate.getHours() - 24)
+      break
+    case TimeRangeItem.week:
+      gteDate.setHours(gteDate.getHours() - 7 * 24)
+      break
+    case TimeRangeItem.month:
+      gteDate.setHours(gteDate.getHours() - (31 * 24))
+      break
+  }
+  return { gteDate: gteDate }
+}
```

Comment:
na praci s Date doporucuji spise pouzivat [date-fns](https://date-fns.org/)... pak to muze vypadat takto

```ts
case TimeRangeItem.now:
  return add(new Date(), { hours: -2 })
```

Reply:
💡